src/analysis/destruct.ml

```diff
@@ -571,9 +590,9 @@ let rec node config source selected_node parents =
             | [ more_precise ] ->
               (* If only one pattern is generated, then we're only refining the
                 current pattern, not generating new branches. *)
+              let punning_field = find_field_name_for_punning_field patt parents in
               let ppat = filter_pat_attr (Untypeast.untype_pattern more_precise) in
-              let str = Mreader.print_pretty
-                  config source (Pretty_pattern ppat) in
+              let str = print_pretty ?punning_field config source (Pretty_pattern ppat) in
               patt.Typedtree.pat_loc, str
             | sub_patterns ->
```

Comment:
Does the issue happens only when there is only one pattern generated ?

It might be worth adding a test case for when there are several, maybe:

```
 $ $MERLIN single case-analysis -start 2:9 -end 2:9 \
  > -filename main.ml <<EOF
  > type t = {a: int option; b: string}
  > let f ({a; b} : t) = assert false
  > EOF
```

Reply:
I have add your test and it seems that this the right behaviour. WDYT?

Reply:
I think the result is what I was expecting :-)